app/components/AccountStateButton.tsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+
+import { PropsWithChildren } from 'react';
+
+import { useAccount } from 'wagmi';
+
+import { ConnectButton } from '@/app/components';
+import { useUnsupportedNetwork } from '@/hooks';
+import { Button } from '@swapr/ui';
+
+export const AccountStateButton = ({ children }: PropsWithChildren) => {
```

Comment:
I'm not a fan of adding state to the name, `AccountButton`, sounds better and implies relationship with account. 

And also not sure if is account is the right name, as I read it, I associate more with the Navbar connect button, that's for me is account. Ability to connect and showing address/balance when connected. 

<img width="199" alt="image" src="https://github.com/user-attachments/assets/72d1acd5-ef77-4d3f-8b8b-194ea06c5add" />

So we probably need to find a better name. 

The case for this component is that we want to make it easier for the user to login if he is not connected, or change to the right network if the user is on different one.

these methods: `isDisconnected`, `unsupportedNetwork`

So we add that capability to extend the button.
I'll give some suggestion, not sure if is good but I think they are better describing:

`ButtonWithConnect` (the idea here it to imply that that's a button has connect capabilities)